heaps/min_heap.py

```diff
@@ -10,67 +10,97 @@ def __str__(self):
     def __repr__(self):
         return str(self.value)
 
+
 class MinHeap:
 
     def __init__(self):
         self.store = []
 
-
-    def add(self, key, value = None):
+    def add(self, key, value=None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(log n)
+            Space Complexity: O(1)
         """
-        pass
+        if value == None:
+            value = key
+
+        self.store.append(HeapNode(key, value))
+        self.heap_up(len(self.store) - 1)
 
     def remove(self):
```

Comment:
👀 Nice work! However space complexity here is O(log n) because of the recursive call stack in `heap_down`